pkg/proxy/proxy.go

```diff
@@ -188,29 +192,67 @@ func (p *Proxy) forwardRequest(host string, w http.ResponseWriter, req *http.Req
 	proxyReq.URL.Host = host
 	proxyReq.URL.Scheme = "https"
 
-	log.Debug("Forwarding request to %s", proxyReq.URL.String())
-	client := http.Client{}
-	resp, err := client.Do(proxyReq)
-	if err != nil {
-		if urlErr, ok := err.(*url.Error); ok && urlErr.Timeout() {
-			writeJSONError(w, http.StatusGatewayTimeout, urlErr)
-		} else {
+	for {
```

Comment:
Do we have a mechanism to prevent infinite redirect loops here?

Reply:
We have the context timeout at https://github.com/teslamotors/vehicle-command/pull/315/files#diff-b41d02f71c670ff801d99892d32ef6dd36e43fcdf4c12c044cd34a073cc18200R249 but we can add a retry count as well.

Reply:
Added a attempt count